### DISCUSSION TITLE
Database Location?

### DISCUSSION THREAD

**Discussion by 6a864c7177** [edrlab/thorium-reader#1932] [Q&A] [answered]
To track assets and progress, I assume that Thorium uses a database, though I suppose it could be a flat file.

Regardless, my question is, what is that mechanism, and where is it stored?

Thanks!

**Comment by danielweck** [comment, marked as answer]
The "app data" `EDRLab.ThoriumReader` folder contains imported publications, parser cache, user configuration, Chromium data, etc. The actual location on the filesystem depends on the operating system.
* On Mac OS: `/Users/{MY_USERNAME}/Library/Application Support/EDRLab.ThoriumReader/` (or with the tilde alias: `~/Library/Application Support/EDRLab.ThoriumReader/`)
* On Windows: `C:\Users\{MY_USERNAME}\AppData\Roaming\EDRLab.ThoriumReader\` (or with the system alias: `%APPDATA%\Roaming\EDRLab.ThoriumReader\`). Note that Windows Store installations also use `C:\Users\{MY_USERNAME}\AppData\Local\Packages\EDRLab.ThoriumReader_{STORE_ID}\LocalCache\Roaming\EDRLab.ThoriumReader\`
* On Linux, `$XDG_CONFIG_HOME` or `~/.config`

See https://www.electronjs.org/docs/latest/api/app#appgetpathname

**Reply by 6a864c7177** [reply]
_Very_ helpful, thank you!!
